src/components/svg-canvas-graph.tsx

```diff
@@ -60,40 +60,21 @@ const SvgCanvas = () => {
         setOffset({ x, y });
 
         dispatch(setActive(node));
-
-        if (!e.shiftKey) {
-            // no shift key -> non multiple selection case
-            if (!selected.has(node)) {
-                // set the current as the only one no matter what the previous selected were
-                dispatch(setSelected(new Set<StnId | MiscNodeId>([node])));
-            } else {
-                // no-op as users may drag the previously selected node(s) for the current selected
-            }
-        } else {
-            // shift key pressed -> multiple selection case
-            if (selected.has(node)) {
-                // remove current if it is already in the multiple selection
-                dispatch(removeSelected(node));
-            } else {
-                // add current in the multiple selection
-                dispatch(addSelected(node));
-            }
-        }
         // console.log('down ', graph.current.getNodeAttributes(node));
     });
     const handlePointerMove = useEvent((node: StnId | MiscNodeId, e: React.PointerEvent<SVGElement>) => {
         const { x, y } = getMousePosition(e);
 
         if (mode === 'free' && active === node) {
-            selected.forEach(s => {
-                if (graph.current.hasNode(s)) {
```

Comment:
Can't we just `if (graph.current.hasNode(s) && s === active) {` ?

Reply:
The node isn't in selected while moving.

I put the "adding node to selected" into "onPointerUp", as the right sidebar will appear and cover the canvas on mobile devices.

Reply:
Fare enough. Let's simplify this to `[selected.has(active) ? undefined : active, ...selected]`.

Reply:
I suddenly think it should be `(selected.has(active) ? selected : [active])`.

- move one node => no selected, do only for `active`.
- move many node => nodes are in `selected`, loop in `selected` .

Reply:
If you only move `[active]`, other selected in the multiple selections will not be moved, right?

Reply:
They will all be moved, because `selected` has `active` .

Reply:
Then how about when I already have multiple selections on stn1 and stn2 and then move stn3 with the shift key pressed? In the main branch, the stn3 should be moved as well.

Reply:
I'll rollback these changes, and just hide the panel when active is not `undefined` to make sure no panel covering the canvas while moving nodes.